src/util/StringUtils.h

```diff
@@ -508,26 +508,30 @@ J join(const vector<J>& to_join, const S& joiner) {
 
 // _____________________________________________________________________________
 inline string lstrip(const string& text, char c) {
-  size_t i = 0;
-  while (i < text.size() && text[i] == c) {
-    ++i;
+  auto it = text.begin();
```

Comment:
You could also add an additional variable `auto right = text.end();` and use that instead of both calls to `text.end()`.

Reply:
I'd hope that every self respecting compiler inlines `std::string::end()` and then it's just a pointer/size comparison either way.

Reply:
Ah yeah, compiler magic. Okay, then leaving it that way could also result in better readable code.